Update JAX FAQ.rst, jax.device_put (#3496)
jax-ml/jax: docs/faq.rst

device and the results are uncommitted on the default device.

Data can also be placed explicitly on a device using :func:`jax.device_put`

> with a ``device`` parameter, in which case it—the data—becomes **committed** to the device:

Reviewer: ```suggestion
with a ``device`` parameter, in which case the data becomes **committed** to the device:
```

Author: ✓ Awesome

>>> import jax
>>> from jax import device_put
>>> print(device_put(1, jax.devices()[2]).device_buffer.device())  # doctest: +SKIP
gpu:2

Computations involving some committed inputs will happen on the
committed device and the result will be committed on the
same device. You should not invoke an operation on

> arguments that are committed to more than one device—it will lead to errors.

Reviewer: Maybe:
Invoking an operation on arguments that are committed to more than one device will raise an error.

WDYT? "It will lead to errors" implies to me that something will eventually go wrong down the line, but I believe this should immediately raise a hopefully-clear error.

Author: I agree ⚡︎

You can also use :func:`jax.device_put` without a ``device`` parameter. In this case
the data is left as is if it's already on the device (committed or not), or a Python 

> value—that is not on any device—is placed uncommitted on the default device.

Reviewer: How about:

You can also use :func:`jax.device_put` without a ``device`` parameter. If the data is already on a device (committed or not), it's left as-is. If the data isn't on any device (i.e. it's a regular Python or numpy value), it's placed uncommitted on the default device.

Author: ⚡️

> Jitted functions behave like any other primitive operations—they will follow the 
> data and will show errors if invoked on data committed on more than one device.

Reviewer: ```suggestion
Jitted functions behave like any other primitive operations: they follow the 
data and raise an error if invoked on data committed on more than one device.
```

Author: 👍

(As of April 2020, :func:`jax.jit` has a `device` parameter

> that affects the device placement slightly. That parameter

Reviewer: ```suggestion
that affects the device placement. That parameter
```

Author: 👍

If you are getting an error that a library function is called with
*"Abstract tracer value encountered where concrete value is expected"*, you may need to

> change how you invoke JAX transformations. Below, you'll first go over such example and then
> a couple of possible solutions. Then, you'll learn the details of what is actually

Reviewer: ```suggestion
change how you invoke JAX transformations. Below is an example and
a couple of possible solutions, followed by the details of what is actually
```

Author: So much better. Thanks!
